jetty-webapp/src/main/java/org/eclipse/jetty/webapp/WebInfConfiguration.java

```diff
@@ -642,8 +642,14 @@ public void unpack (WebAppContext context) throws IOException
                     }
                     else
                     {
+                        long lastModified;
+                        // read last modified without caching a war file
+                        // so that webapp can be redeployed on Windows
+                        try (Resource r = Resource.newResource(web_app.getURL().toExternalForm(), false)) {
```

Comment:
`URL.toExternalForm()` does not escape properly, please use the URI version.

Also, why are we doing it this way?
Wouldn't it just be ...

``` java
long lastModified = web_app.lastModified();
```

... without the extra `Resource` instantiation?

Reply:
Also, this seems like the wrong place to be making this kind of change.

Why not make it in `JarFileResource#lastModified()` instead?

Reply:
> `URL.toExternalForm()` does not escape properly, please use the URI version.

`Resource.newResource`, `JarResource.copyTo` used 5 lines below, `UrlResource.toString` used in `JarResource.newJarResource` called below, they all call `URL.toExternalForm()`, so I figured that was the right and consistent way. 

> Why not make it in JarFileResource#lastModified() instead?

I'm not sure what's the proper way of making that change there. 

```
    at java.util.zip.ZipFile.<init>(ZipFile.java:150)
    at java.util.jar.JarFile.<init>(JarFile.java:166)
    at java.util.jar.JarFile.<init>(JarFile.java:103)
    at sun.net.www.protocol.jar.URLJarFile.<init>(URLJarFile.java:93)
    at sun.net.www.protocol.jar.URLJarFile.getJarFile(URLJarFile.java:69)
    at sun.net.www.protocol.jar.JarFileFactory.get(JarFileFactory.java:94)
    at sun.net.www.protocol.jar.JarURLConnection.connect(JarURLConnection.java:122)
    at sun.net.www.protocol.jar.JarURLConnection.getJarFile(JarURLConnection.java:89)
    at org.eclipse.jetty.util.resource.JarFileResource.newConnection(JarFileResource.java:129)
    at org.eclipse.jetty.util.resource.JarResource.checkConnection(JarResource.java:74)
    at org.eclipse.jetty.util.resource.JarFileResource.checkConnection(JarFileResource.java:96)
    at org.eclipse.jetty.util.resource.JarFileResource.lastModified(JarFileResource.java:255)
```

Are you proposing to set `_useCaches` to `false` in the beginning of `JarFileResource#lastModified` method and then revert it back to its original state and close connection in finally?